### PR TITLE
Move CSparse functions to separate package, take two

### PR DESCRIPTION
**Pull request by Sacha0**
This pull request is a rework of #16070 and addresses #12231. See the former for discussion. Best!

### CODE REVIEW

File: base/deprecated.jl [hunk 1]
```diff
@@ -769,6 +769,25 @@ end
 @deprecate slice view
 @deprecate sub view
 
+# Point users to SuiteSparse
+function ereach{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, k::Integer, parent::Vector{Ti})
+    error(string("ereach(A, k, parent) now lives in package SuiteSparse.jl. Run",
+        "Pkg.Add(\"SuiteSparse\") to install SuiteSparse on Julia v0.5."))
```
Comment (tkelman) [inline base/deprecated.jl]: lowercase `a`

Comment (Sacha0) [inline base/deprecated.jl]: Fixed. Thanks!

**Comment by Sacha0** [comment]
Thanks for the review and merge!

**Comment by KristofferC** [comment]
Thanks for the help.

**Comment by Eetion** [comment]
Did this Csparse refer to a wrapper for the C code, or to `Csparse.jl`?  If the latter, does anyone know if there's anyone still maintaining a pure Julia implementation of Csparse?

**Pull request merged**